scripts/collect_ad_boilerplate.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+
+import logging
+
+
+def write_detector_class(boilerplate_file, dev_name, det_name, cam_name):
+    """
+    Writes boilerplater 'Detector' class for ophyd/areadetector/detectors
+    """
+
+    boilerplate_file.write(
+        f'''
+class {det_name}(DetectorBase):
+    _html_docs = ['{dev_name}Doc.html']
+    cam = C(cam.{cam_name}, 'cam1:')
+''')
+
+
+def parse_pv_structure(driver_dir):
+    """
+    Reads all .template files in the specified driver directory and maps them
+    to the appropriate EPICS signal class in ophyd
+
+    Also determines if the Cam class should extend the FileBase class as well.
+    """
+
+    # Find the template directory following the standard areaDetector project format
+    template_dir = driver_dir
+    for dir in os.listdir(driver_dir):
+        if dir.endswith('App'):
+            template_dir = os.path.join(template_dir, dir, 'Db')
+            break
+    logging.debug(f'Found template dir: {template_dir}')
+
+    # Create a list of file paths to all template files.
+    template_files = []
+    for file in os.listdir(template_dir):
+        if file.endswith('.template'):
+            template_files.append(os.path.join(template_dir, file))
+            logging.debug(f'Found template file {file}')
+
+    # Dict mapping pv name to appropriate EPICS signal, based on typical
+    # PV and PV_RBV format for areaDetector
+    pv_to_signal_mapping = {}
+    include_file_base = False
+
+    for file in template_files:
+        logging.debug(f'Collecting pv info from {os.path.basename(file)}')
+        fp = open(file, 'r')
```

Comment:
I'd recommend using the context manager here, so that you don't have e to worry about closing the file.
```py
with open(file, 'r') as fp:
    ...
```